### PR TITLE
[cms-v2][dcc-v2] New DCC v2

### LINKED ISSUE #1668: [cms] Submit DCC enhancement
When selecting either Issuance or Revocation the form should change accordingly.

Issuance: 
- Show Domain
- Show Contractor Type
- Dropdown showing only Nominee Users

Revocation:
- Hide Domain (automatically set to whatever user is selected)
- Hide Contractor Type (since this should be set to Revoke internally)
- Dropdown showing only Full Users

### LINKED ISSUE #1670: [cms] DCC Sponsor statement length warning/display
I recently updated the sponsor statement to now have 5000 character allowance. 

There should be a display shown X/5000 characters shown so people may have an idea how much they have left to write in the text area.

### LINKED ISSUE #1686: [cms-v2] New DCC
The submit/new DCC page for CMS v2 should include the same info than v1, such as:

- *DCC Type*
    Issuance or revocation
- *Nominee User ID*:
    This should include the adjustments from #1669 
- *Statement*
- *Domain*
-*Contractor Type*

The enhancements from #1668 and #1670 should also be included.

### PR DESCRIPTION
**Pull request by victorgcramos**
This diff adds the New DCC module for cmsv2, including:

- [x] Nominee user selector popoulated with proper users (#1668)
- [x] Dcc type selector
- [x] Contractor Type selector
- [x] Statement field with max length validator (#1670)
- [x] Domain selector
- [x] Form Validation

Closes #1686 
Closes #1670 
Closes #1668

### CODE REVIEW

File: src/componentsv2/DccForm/DccForm.jsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
+import React, { useMemo, useState, useCallback, useEffect } from "react";
+import PropTypes from "prop-types";
+import { BoxTextInput, Button, Message, RadioButtonGroup, classNames } from "pi-ui";
+import { Formik } from "formik";
+import { withRouter } from "react-router-dom";
+import usePolicy from "src/hooks/api/usePolicy";
+import { dccValidationSchema } from "./validation";
+import { SelectField } from "src/componentsv2/Select";
+// import DraftSaver from "./DraftSaver";
```
Comment (amassarwi) [inline src/componentsv2/DccForm/DccForm.jsx]: Comment

File: src/pages/RoutesCMS.jsx [hunk 2]
```diff
@@ -98,20 +99,24 @@ const Routes = ({ location }) => {
...
+            title="New DCC"
+            exact
+            component={withDcc(PageDccNew)}
+          />
+          <AuthenticatedRoute
+            path="/dccs/:token"
+            title="DCC Detail"
+            exact
+            component={withDcc(PageDccDetail)}
+          />
```
Comment (amassarwi) [inline src/pages/RoutesCMS.jsx]: Let's cleanup all commented out routes

File: src/componentsv2/DccForm/DccForm.jsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      <BoxTextInput
+        placeholder="Statement"
+        name="statement"
+        tabIndex={1}
+        value={values.statement}
+        error={touched.statement && errors.statement}
+        onChange={handleChangeWithTouched("statement")}
+      />
+      <div className="justify-right">
+        {/* <DraftSaver submitSuccess={submitSuccess} /> */}
```
Comment (amassarwi) [inline src/componentsv2/DccForm/DccForm.jsx]: Commented out draft button

File: src/containers/DCC/DccProvider.js [hunk 4]
```diff
@@ -9,3 +9,9 @@ const customConfigDcc = {
 export const DccProvider = ({ children }) => (
   <CustomConfigProvider {...{ children, ...customConfigDcc }} />
 );
+
```
Comment (amassarwi) [inline src/containers/DCC/DccProvider.js]: Other providers are `.jsx` files and not `.js`

**Comment by amassarwi** [comment]
We should clear the user selection on radio button clicks, because each type has different users list - see screen record below:

user-reset

File: src/componentsv2/DccForm/DccForm.jsx (hunk 3 above)
Comment (amassarwi) [reply src/componentsv2/DccForm/DccForm.jsx]: Daft coming next I guess :rofl:

Comment (victorgcramos) [reply src/componentsv2/DccForm/DccForm.jsx]: hahaha sure!!

File: src/componentsv2/DccForm/DccForm.module.css [hunk 5]
```diff
@@ -13,3 +13,7 @@
   min-height: 2rem;
   margin-top: 0.3rem;
 }
+
+.radioButton {
+  padding-bottom: 1em;
```
Comment (amassarwi) [inline src/componentsv2/DccForm/DccForm.module.css]:
```suggestion
  padding-bottom: 1rem;
```

**Comment by amassarwi** [comment]
Submit button is disabled after refresh even though values are valid (I had something similar in the invoice form, checkout my approach [there](https://github.com/decred/politeiagui/blob/cmsv2/src/componentsv2/InvoiceForm/InvoiceForm.jsx#L229)) 

image

**Pull request merged**